src/HealthChecks.Nats/NatsHealthCheck.cs

```diff
@@ -0,0 +1,101 @@
+using System.Text;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using NATS.Client;
+
+namespace HealthChecks.Nats
+{
+    /// <summary>
+    /// Health check for Nats Server
+    /// </summary>
+    public sealed class NatsHealthCheck : IHealthCheck, IDisposable
+    {
+        /// <summary>
+        ///  Provides factory methods to create connections to NATS Servers.
+        /// </summary>
+        private static readonly ConnectionFactory _connectionFactory = new ConnectionFactory();
+
+        private readonly NatsOptions _options;
+
+        /// <summary>
+        /// An NATS.Client.IConnection object connected to the NATS server.
+        /// </summary>
+        private IConnection? _connection = null;
+
+        public NatsHealthCheck(NatsOptions natsOptions)
+        {
+            _options = natsOptions;
+        }
+
+        public Task<HealthCheckResult> CheckHealthAsync(
+            HealthCheckContext context,
+            CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                _connection = CreateConnection(_options);
+                if (_connection is null)
+                    throw new ArgumentNullException(nameof(_connection));
+                var healthCheckResult = GetHealthCheckResultFromState(_connection);
+                return Task.FromResult(healthCheckResult);
+            }
+            catch (Exception ex)
+            {
+                var unhealthy = new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
+                return Task.FromResult(unhealthy);
+            }
+
+            IConnection CreateConnection(NatsOptions options)
+            {
+                if (options == null)
+                    throw new ArgumentNullException(nameof(options));
```

Comment:
Move this check into ctor or just remove it.

Reply:
Commit: 0ab4b8af223791d4d882a83a2d82eaa9f0ab8cc9 [0ab4b8a]
Date: Tuesday, March 1, 2022 13:32:27